plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -332,8 +347,8 @@ public String getPluginDownloadUrl(Plugin plugin) {
 
             groupId = groupId.replace(".", "/");
 
-            String incrementalsVersionPath = new StringBuffer(incrementalsVersion).append("/").append(pluginName)
-                    .append("-").append(incrementalsVersion).append(".hpi").toString();
+            String incrementalsVersionPath = new StringBuffer(pluginName).append("/").append(incrementalsVersion).append("/").
```

Comment:
This would probably be clearer as a formatted string.
Also is this the correct name? path seems to be doubled up?
```suggestion
            String incrementalsVersionPath = String.format("%s/%s/%s-%s.hpi", pluginName, incrementalsVersion, pluginName, incrementalsVersion);
          
```